client/src/store/utils/thunkCreators.js

```diff
@@ -67,13 +68,29 @@ export const logout = (id) => async (dispatch) => {
   }
 };
 
+const findLastReadMessageId = (messages, userId) => {
+  let lastReadMessageId = -1;
+  messages.forEach(message=>{
+    if (message.id > lastReadMessageId && message.read === true && message.senderId === userId)
+      {
+        lastReadMessageId = message.id
+      }
+  })
+  return lastReadMessageId === -1 ? undefined: lastReadMessageId;
+};
```

Comment:
This logic could be implemented using `.find()`

Reply:
if you notice, we are updating the value of `lastReadMessageId` on each iteration of the loop. The `find` method would only return one occurrence of it. Whereas we need to check all occurrence and make sure we return the maximum one only (and we assume here that message id's could potentially be unsorted) . I don't see how to achieve that using find. If you have an input on the specifics, feel free to share